### ISSUE TITLE
bug: Missing slotted content when slot is forwarded from a non-shadow component

### ISSUE THREAD

**Issue by asik** [ionic-team/stencil#4525]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.0.0 (also seen in 3.4.0)

### Current Behavior

I'm not sure exactly what the problem is here, but basically I have a component with this structure:

```jsx
@Component({
  tag: 'non-shadow-host',
  shadow: false,
})
export class NonShadowHost {
  render() {
    return (
      <div>
        <drop-down>
          <slot name="main-content" slot="main-content" />
          <div slot="dropdown-content-element">
            <drop-down-content>
              <slot name="dropdown-content" />
            </drop-down-content>
          </div>
        </drop-down>
      </div>
    );
  }
}
```

(the name drop-down is meaningless, I stripped out all code related to that). If I set `shadow: true`, this works, if I remove it, the slotted content in "dropdown-content" doesn't render. In the following screenshot, the selected node should get slotted but doesn't, and doesn't show up in the rendered HTML.

![image](https://github.com/ionic-team/stencil/assets/1631157/09765b99-034e-4549-b9b3-a18120fc1f99)

I made a copy of this component, changed the name for shadow-host and set shadow to true, and now it renders as expected.

![image](https://github.com/ionic-team/stencil/assets/1631157/fa16a9f8-e552-4505-b1c7-0e8d6480e6d4)

![image](https://github.com/ionic-team/stencil/assets/1631157/75329f1c-7bef-480c-8341-46804962bc13)

I'm not sure if I'm trying to do something unsupported here, but I didn't see any warnings in the documentation about interop between shadow and non-shadow components.

### Expected Behavior

I would expect both components to end up rendering the same thing on screen.

### Steps to Reproduce

Clone the linked repo
`yarn && yarn start`
Check out example 2 labeled Missing slotted content bug in index.html.

### Code Reproduction URL

https://github.com/asik/stencil-repro

**Comment by rwaskiewicz** [comment]
Thanks @asik! The reproduction case helps immensely! I believe this has something to do with the differing values of `shadow` between `'drop-down-content'` and `'non-shadow-host'`. I've labeled this to get ingested into our backlog and as a part of the larger effort to fix up bugs related to slots in Stencil. Thanks again!

**Comment by rwaskiewicz** [comment]
The fix for this issue has been released as a [part of today's Stencil v4.7.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.7.0). Should the issue reappear, please feel free to open a new ticket. Thanks!
